sdk/storage/storage-file-share/src/Clients.ts

```diff
@@ -2756,6 +2760,78 @@ export class ShareDirectoryClient extends StorageClient {
       span.end();
     }
   }
+
+  /**
+   * Renames a directory.
+   * This API does not support renaming a directory from one share to another, or between storage accounts.
+   *
+   * @param destinationPath - Specifies the destination path to rename to.
+   * @param options - Options for the renaming operation.
+   * @returns Response data for the file renaming operation.
+   *
+   * Example usage:
+   *
+   * ```js
+   *
+   * // Rename the directory
+   * await diretoryClient.create(destinationPath);
+   * console.log("Renamed directory successfully!");
+   * ```
+   */
+  public async rename(
+    destinationPath: string,
+    options: DirectoryRenameOptions = {}
+  ): Promise<{
+    destinationDirectoryClient: ShareDirectoryClient;
+    directoryRenameResponse: DirectoryRenameResponse;
+  }> {
+    const { span, updatedOptions } = createSpan("ShareDirectoryClient-rename", options);
+    const split: string[] = destinationPath.split("?");
+    let destinationUrl: string;
+    if (split.length === 2) {
+      const pathOnly = encodeURIComponent(split[0]);
+      const renameDestination = `/${this.shareName}/${pathOnly}`;
+      destinationUrl = setURLPath(this.url, renameDestination);
+      destinationUrl = setURLQueries(destinationUrl, split[1]);
+    } else if (split.length === 1) {
+      const pathOnly = encodeURIComponent(destinationPath);
+      const renameDestination = `/${this.shareName}/${pathOnly}`;
+      destinationUrl = setURLPath(this.url, renameDestination);
+    } else {
+      throw new RangeError("Destination path should not contain more than one query string");
+    }
+
+    const destDirectory = new ShareDirectoryClient(destinationUrl, this.pipeline);
+
+    try {
+      const response = await destDirectory.context.rename(this.url, {
+        ...updatedOptions,
+        sourceLeaseAccessConditions: updatedOptions.sourceLeaseAccessConditions
+          ? {
+              sourceLeaseId: updatedOptions.sourceLeaseAccessConditions.leaseId,
```

Comment:
no longer to cast now?

Reply:
The auto-generated code uses SourceLeaseAccessConditions with a member sourceLeaseId. We'd need to assign the `LeaseAccessConditions` instance to `SourceLeaseAccessConditions`